openff/models/types.py

```diff
@@ -56,17 +56,23 @@ def validate_type(cls, val):
                     # some custom behavior could go here
                     assert unit_.dimensionality == val.dimensionality
                     # return through converting to some intended default units (taken from the class)
+                    val._magnitude = float(val.m)
                     return val.to(unit_)
                     # could return here, without converting
                     # (could be inconsistent with data model - heteregenous but compatible units)
                     # return val
                 if _is_openmm_quantity(val):
                     return _from_omm_quantity(val).to(unit_)
-                if isinstance(val, (float, int)) and not isinstance(val, bool):
+                if isinstance(val, int) and not isinstance(val, bool):
+                    # coerce ints into floats for a FloatQuantity
+                    return float(val) * unit_
+                if isinstance(val, float):
                     return val * unit_
                 if isinstance(val, str):
                     # could do custom deserialization here?
-                    return unit.Quantity(val).to(unit_)
+                    val = unit.Quantity(val).to(unit_)
+                    val._magnitude = float(val._magnitude)
```

Comment:
@mattwthompson this is a little sharp, but avoids the double creation of Quantity.  Either way is fine